Library/ViewModels/ProjectPageViewModelTests.swift

```diff
@@ -1231,6 +1206,36 @@ final class ProjectPageViewModelTests: TestCase {
     self.updateDataSourceProjectProperties.assertDidEmitValue()
   }
 
+  func testOutput_UpdateDataSourceProjectProperties_ReloadsAfterUserSessionStarted() {
+    let overviewSection = NavigationSection.overview.rawValue
+    let environmentalCommitmentsSection = NavigationSection.environmentalCommitments.rawValue
+
+    withEnvironment(currentUser: nil) {
+      self.vm.inputs.configureWith(projectOrParam: .left(self.projectWithEmptyProperties), refTag: .category)
+
+      self.updateDataSourceProjectProperties.assertDidNotEmitValue()
+
+      self.vm.inputs.viewDidLoad()
+
+      self.updateDataSourceProjectProperties.assertDidNotEmitValue()
+
+      self.vm.inputs.projectNavigationSelectorViewDidSelect(index: overviewSection)
+
+      // The view model skips the first emission
+      self.updateDataSourceProjectProperties.assertDidNotEmitValue()
+
+      self.vm.inputs.projectNavigationSelectorViewDidSelect(index: environmentalCommitmentsSection)
+
+      self.updateDataSourceProjectProperties.assertDidEmitValue()
+
+      withEnvironment(currentUser: .template) {
+        self.vm.inputs.userSessionStarted()
+
+        self.updateDataSourceProjectProperties.assertDidEmitValue()
```

Comment:
Just curious, does this means the observer emits again, or if it emitted once, it's still considered `emittted` the second time the method is called?

Reply:
I believe this implies it is emitted again although I can see how this is a bit ambiguous and not entirely obvious. The gist of the test was to verify that after the user session starts that emits so I think it is on the right page?

Reply:
yep, its' probably that it emits again, I'll double check today, not 100% on it, otherwise everything else is gtg.